pkg/gateway/operations/postobject.go

```diff
@@ -149,6 +149,18 @@ func (controller *PostObject) Handle(w http.ResponseWriter, req *http.Request, o
 	// POST is only supported for CreateMultipartUpload/CompleteMultipartUpload
 	// https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html
 	// https://docs.aws.amazon.com/AmazonS3/latest/API/API_CompleteMultipartUpload.html
+	branchExists, err := o.Catalog.BranchExists(req.Context(), o.Repository.Name, o.Reference)
```

Comment:
1. question- this is relevant only for create multipart?
2. do we like to push this code down one level to get the operation stat?

Reply:
1. That's correct. Indeed moved the branch check to the create multipart after getting the stats.

Reply:
Good point. We indeed only need to double check before the create multipart phase and not in the rest of the handle function. Fixed this by moving the check to HandleCreateMultipartUpload function specifically.